Add missing MariaDB Grants to mysql module

cebe opened the pull request:
### What does this PR do?

MariaDB has added some grants in 10.4.x and 10.5.x that are not present here, which results in an error when creating.

Also improved exception handling in `grant_add` which did not log the original error message and replaced it with a generic error.


### What issues does this PR fix or reference?

Fixes: #61409 

### Previous Behavior

Error when trying to add grants that salt is not aware of.
The error message is very unspecific: "Error during grant generation"

### New Behavior

No error for new MariaDB grants, better error message in case of unsupported grants.


### Merge requirements satisfied?

<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated (there are no tests to update for this change)

### Commits signed with GPG?

Yes

cebe committed:
Add missing MariaDB Grants to mysql module

MariaDB has added some grants in 10.4.x and 10.5.x that are not present here, which results in an error when creating.

This is an addition to #59280.

Also improved exception handling in `grant_add` which did not log the original error message and replaced it with a generic error.

cebe commented on salt/modules/mysql.py:
```diff
@@ -107,16 +113,22 @@
+    "REPLICA MONITOR",  # MariaDB since 10.5.9
     "REPLICATION CLIENT",
+    "REPLICATION MASTER ADMIN",  # MariaDB since 10.5.2
+    "REPLICATION REPLICA",  # MariaDB since 10.5.1
```
Just found that it is not possible to apply this grant as it is an alias for `REPLICATION SLAVE` and the grant reported by MariaDB is `REPLICATION SLAVE` so when trying to apply `REPLICATION REPLICA` it fails because it is detected as a mismatch...

So to fully fix this it is required to add the information about grant aliases in this module.

ITJamie replied:
https://github.com/saltstack/salt/pull/59280 would be a good long term fix for this.

However i still think this pr would be good to merge now as it would at least restore functionality to actually use grants. It would be on the user to know which grants to use based on version until something like the above PR could be reworked

ITJamie commented:
another issue for this: https://github.com/saltstack/salt/issues/58297

garethgreenaway commented:
@cebe Looks good.  Would be great to add some tests, please let me know if you need help getting those in.

Ch3LL commented:
bump @cebe can you respond to @garethgreenaway comment above?

garethgreenaway commented:
https://github.com/saltstack/salt/pull/59280 has been merged in which addresses similar issues to this PR.

Ch3LL commented:
@cebe can we close this PR In favor of https://github.com/saltstack/salt/pull/59280?

garethgreenaway commented:
Closing this one out in favor of https://github.com/saltstack/salt/pull/59280

cebe commented:
> @cebe can we close this PR In favor of #59280?

#59280 does not contain all of what is in here, I'll update my branch with the changes from #59280 to show the diff. @garethgreenaway Please reopen.

Ch3LL commented:
@cebe can you resovle the merge conflict?

@garethgreenaway will this require tests or is there already test coverage for a change like this?

cebe commented:
@Ch3LL I just rebased my commit on master, here are the changes that are left:

- #59280 was missing is the `DELETE HISTORY` grant and I add my comments about MariaDB version
- I changed the exception handling to log the actual error when grant changes failed
- #59280 did not contain a changelog, should I keep the changelog with my PR as file number or change the file to 59280?

as far as I see, no extra tests are needed for this.

garethgreenaway committed:
Merge branch 'master' into patch-1

garethgreenaway commented:
@Ch3LL I think we're good on existing test coverage.

@cebe The changelog is correct, the other PR didn't have one but we added one after the fact. Tests are failing for unrelated reasons, once they're passing we'll get this merged.

cebe commented:
Nice, thanks for merging!
